test/gtest/ucs/test_sock.cc

```diff
@@ -148,148 +138,127 @@ UCS_TEST_F(test_socket, sockaddr_get_port) {
         socket_err_exp_str = "unknown address family:";
         scoped_log_handler log_handler(socket_error_handler);
 
-        port = sin_port;
-        EXPECT_EQ(UCS_ERR_INVALID_PARAM,
-                  ucs_sockaddr_get_port((const struct sockaddr*)&sa_un, &port));
+        port             = sin_port;
+        saddr->sa_family = AF_UNIX;
+        EXPECT_EQ(UCS_ERR_INVALID_PARAM, ucs_sockaddr_get_port(saddr, &port));
         /* Check that doesn't touch provided memory in error case */
         EXPECT_EQ(sin_port, port);
     }
 }
 
 UCS_TEST_F(test_socket, sockaddr_get_inet_addr) {
-    struct sockaddr_un sa_un = {};
-    struct sockaddr_in sa_in;
-    struct sockaddr_in6 sa_in6;
+    struct sockaddr_storage ss = {0};
+    struct sockaddr *saddr     = (struct sockaddr*)&ss;
     struct in_addr sin_addr;
     struct in6_addr sin6_addr;
 
-    sa_in.sin_family   = AF_INET;
-    sa_in6.sin6_family = AF_INET6;
-    sa_un.sun_family   = AF_UNIX;
-
-    sin_addr.s_addr = sa_in.sin_addr.s_addr = htonl(INADDR_ANY);
-    sin6_addr       = sa_in6.sin6_addr      = in6addr_any;
-
     /* Check with wrong IPv4 */
     {
-        EXPECT_EQ(&sa_in.sin_addr,
-                  ucs_sockaddr_get_inet_addr((const struct sockaddr*)&sa_in));
-        EXPECT_EQ(0, memcmp(&sa_in.sin_addr, &sin_addr,
-                            sizeof(sa_in.sin_addr)));
+        saddr->sa_family                              = AF_INET;
+        sin_addr.s_addr                               = htonl(INADDR_ANY);
+        ((struct sockaddr_in*)saddr)->sin_addr.s_addr = htonl(INADDR_ANY);
+        EXPECT_EQ(&((struct sockaddr_in*)saddr)->sin_addr,
+                  ucs_sockaddr_get_inet_addr(saddr));
+        EXPECT_EQ(0, memcmp(&((struct sockaddr_in*)saddr)->sin_addr, &sin_addr,
+                            sizeof(sin_addr)));
```

Comment:
maybe add helper var
`struct sockaddr_in *saddr_in = (struct sockaddr_in*)saddr;`
and same for in6